docs/reference/search/search.asciidoc

```diff
@@ -1,13 +1,233 @@
 [[search-search]]
 === Search
 
-The search API allows you to execute a search query and get back search hits
-that match the query. The query can either be provided using a simple
+Returns search hits that match the query defined in the request.
+
+[source,js]
+--------------------------------------------------
+GET /twitter/_search?q=user:kimchy
+--------------------------------------------------
+// CONSOLE
+// TEST[setup:twitter]
+
+
+[[search-search-api-request]]
+==== {api-request-title}
+
+`GET /{index}/_search` +
+
+`GET /all/_search`
+
+
+[[search-search-api-desc]]
+==== {api-description-title}
+
+Allows you to execute a search query and get back search hits that match the 
+query. The query can either be provided using a simple
 <<search-uri-request,query string as a parameter>>, or using a
 <<search-request-body,request body>>.
 
+[[search-partial-responses]]
+===== Partial responses
+
+To ensure fast responses, the search API will respond with partial results if 
+one or more shards fail. See <<shard-failures, Shard failures>> for more 
+information.
+
+[[search-search-api-path-params]]
+==== {api-path-parms-title}
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=index]
+
+
+[[search-search-api-query-params]]
+==== {api-query-parms-title}
+
+`allow_no_indices`::
```

Comment:
You can replace this def with `include::{docdir}/rest-api/common-parms.asciidoc[tag=allow-no-indices]`

Added with #45699